### ISSUE TITLE
Some aggressive easy preys should not aggro high level players.

### ISSUE THREAD

**Issue by beauxq** [DerpyProjectGroup/topaz#687]
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened
- [x] I have checked the commit log to see if the issue has been resolved since my server was last updated
- [x] I have read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I have specified what branch this happens on branch: base

**_Additional Information_** (Steps to reproduce/Expected behavior) **:** 

------

Some aggressive easy preys should not aggro high level players.

High levels (75+) should be able to wander freely without aggro around the lower level parts of Quicksand Caves, and the lower level parts of Temple of Uggalepih, and other places, where all of the mobs are below level 60.

The Tonberries and Anticans and all other normal aggressive mobs in the range 56-59 should check as easy prey, but should not aggro.

------

from:
https://ffxiclopedia.fandom.com/wiki/Aggro

> Monsters that check as "Too Weak" will generally not be aggressive, along with some monsters in the "Easy Prey" range.

> monster spawns with a level range reaching 61+ will continue to aggro up to and including level 99. This is irrespective of the particular monster's level; a level 59 monster in a 59-61 range spawn will aggro, but a level 60 in a 58-60 won't.

------

That isn't very precise, and doesn't make sense for describing what happens below level 75.
It talks mostly in terms of what does aggro, and doesn't say much about what doesn't aggro.
But it does clearly say that in some cases, a level 60 will not aggro.

If you try to interpret it as specifying what doesn't aggro, it would seem to be saying that when I'm level 69, a level 51 mob aggros me, but then at level 71 (or 70), a level 60 doesn't aggro me.

I gain 1 or 2 levels, and suddenly 10 levels of mobs stop aggroing me?

I'm pretty sure that that's not the case.

------

This is my hypothesis:

topOfSpawn = the highest level that this mob group could possibly spawn at
ableToAggroAtThisLevel = (! tooWeak) && (topOfSpawn > 60 || playerLevel - topOfSpawn < 15)

------

If this hypothesis is correct, it is possible to find easy preys that don't aggro as low as level 51 (assuming that our experience tables are correct giving exp to a level 51 from a level 36 mob), but it would be extremely rare at level 51, because most level 36 mobs would still aggro. It would become gradually more common as the player's level rises. It wouldn't be noticed by most people until around level 70+.
I think that's why it says "From level 71 (possibly 70 Verification Needed) onwards".

------

How to test this hypothesis:

Take a level 51 character and a level 36 character.
Find some mob groups that have a spawn level range up to but not above level 36.
(Don't just trust the level ranges on the wiki. Look around a lot, and kill a bunch, to make sure none of the mobs ever spawn above level 36. If any of them spawn above 36, then this hypothesis can't be tested there.)
Find one of those mobs that is exactly level 36.
Make sure it checks as easy prey to the level 51. (If it isn't, then our exp tables are wrong, and this hypothesis can't be tested there.)
See if it aggros the level 51 player. If it does, then this hypothesis is incorrect.

**Comment by beauxq** [comment]
I have seen this https://github.com/LandSandBoat/server/issues/25
and this is not the same issue.

Even if all of the checking were perfect and correct, this aggro would still be an issue.

**Comment by TeoTwawki** [comment]
> I gain 1 or 2 levels, and suddenly 10 levels of mobs stop aggroing me?
>
> I'm pretty sure that that's not the case.

ah yes, `#shitWikiSays`

makes for tons of fun having to reconfirm things on retail, usually more than once as people read it and then check again

**Comment by beauxq** [comment]
And in case anyone wants to test whether this is a real issue,
that's much easier to test on retail:

As level 75+, enter Quicksand Caves from Western Altepa Desert (J-9).

There you will see lots of Antican that are ALL easy prey, and NONE of them aggro a 75+.

(They're all in the 56-59 range.)

**Comment by TeoTwawki** [comment]
good news is we can pull their exact levels during this as well (widescan packet) so then we know if whole ranges get their agro "shut off" rather than a hard cut off.

**Comment by RAIST5150** [comment]
The very scenario you mentioned of a mob going from passive to aggro after gaining a few levels did in fact happen in retail back in the day (WotG and SoA era).  It was quite the WTF moment for a lot of people.

IDK if SE ever tweaked it or not, but when I was doing stuff here in QSC, Ugly Pee, Pso Xja, Zvahl, etc... mobs behaved pretty much as I remember them doing back then (circa 2012/2013.

**Comment by beauxq** [comment]
For retail testing on the range cut off vs the level cut off, these might be useful:

https://ffxiclopedia.fandom.com/wiki/Esbat 59 - 61 Uleguerand Range
so 59 Esbat should aggro

https://ffxiclopedia.fandom.com/wiki/War_Smilodon 57 - 60 Jugner Forest (S)
so 60 War Smilodon should not aggro

But there is the possibility that the level range information is incorrect, so that would need to be verified too.
(Make sure there are no War Smilodons ever spawning at 61, and make sure there are Esbats spawning at 61.)

**Comment by ShiyoKozuki** [comment]
You just have to edit exp_table.sql and remove a LOT of things from giving certain levels xp to fix this. It has level 15 chars getting exp from stuff that's like -15 which is really wrong.

**Comment by beauxq** [comment]
> You just have to edit exp_table.sql and remove a LOT of things from giving certain levels xp to fix this.

No, that is wrong. One of the main points of this is that aggro should not be based on xp.

**Comment by ShiyoKozuki** [comment]
It's how LSB is coded, so making them too weak stops aggro. On retail past char level 75 certain mobs stay EP(incredibly easy prey) but won't aggro.

For anyone who takes up this task, if a group of mobs level range is 30-33 with 30-32 being too weak and 33 is easy prey, on retail the 33 won't aggro you either. It's based on the level ranges minimum level.

**Comment by beauxq** [comment]
> It's based on the level ranges minimum level.

Where did you get that information? ffxiclopedia says it's based on the level range's maximum level.
